### DISCUSSION TITLE
twitch plays bot

### DISCUSSION THREAD

**Discussion by twitchplayskh** [howroyd/simonsays#43] [Ideas]
so i have a custom twitch plays bot but its a bit old now and my dev friend has no time to update it no more https://github.com/deeredman1991/Twitch-Plays-Bot  i was thinking of using this for my next tp event but its missing things that my old bot has like 

. freeze the windows process on and off to play a input then freeze till next input is made  good for 3d games how it work https://www.twitch.tv/cosmicblip/clip/TangentialInspiringVanillaAMPTropPunch
. string cmd so i can put a string of cmds like up don left right  and it plays then one by one in a line 

not a must but cool 
. game pad support we made are bot just work with game pads for better compatibility  with emu and lots of pc games  

that's all i think it would need so i can use it on lots of games

**Comment by howroyd** [comment]
Super interesting, thanks for the suggestions.

The windows freezing and controllers/joysticks are things I had not considered.  The latter I don't think would be hard to do, I currently have abstraction for keyboard vs mouse, so adding other input devices should be do-able with minimal fuss.

Window freezing I have no idea about so I shall look at your code and do some research.

Thank you very much; I shall make sure these suggestions get worked on!

**Comment by twitchplayskh** [comment]
> Super interesting, thanks for the suggestions.
> 
> The windows freezing and controllers/joysticks are things I had not considered. The latter I don't think would be hard to do, I currently have abstraction for keyboard vs mouse, so adding other input devices should be do-able with minimal fuss.
> 
> Window freezing I have no idea about so I shall look at your code and do some research.
> 
> Thank you very much; I shall make sure these suggestions get worked on!

awww you are so nice  for the freezing  thing if i remember the program looked for the pid id or program name and just freezing  the windows process haha

**Comment by howroyd** [comment]
Lol nice.  Yeah I shall see if I can do it a bit more elegantly :)

**Comment by twitchplayskh** [comment]
> Lol nice. Yeah I shall see if I can do it a bit more elegantly :)

yes alot of the time it did not work or i could not find the pid id haha  are bot kept rip a lot as well was not good but we could not find what it was to fix hahah

**Comment by twitchplayskh** [comment]
any updates on this?
